### PR TITLE
PLAT-907-2 Eliminated AGSystemModulePermissionAssignment etc.

### PR DESCRIPTION
**Pull request by oliver-richers**

### CODE REVIEW

File: platform-core-module/src/main/java/com/softicar/platform/core/module/access/permission/ConvertedCoreModulePermission.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	@Override
+	public IDisplayString getTitle() {
+
+		return modulePermission.getTitle();
+	}
+
+	@Override
+	public boolean test(R tableRow, IBasicUser user) {
+
+		return AGUser.get(user).hasModulePermission(modulePermission, AGCoreModuleInstance.getInstance().pk());
```
Comment (alexander-schmidt) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/access/permission/ConvertedCoreModulePermission.java:28]: How certain are we about `AGUser.get` yielding non-null, here?

File: platform-integration/src/main/resources/com/softicar/platform/integration/database/structure/version/v20-migration.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
+-- migration from platform-19 to platform-20
+
+-- TODO change ID of core module instance from 1 to 0
+
+-- migrate permission assignments for the core module
+INSERT INTO `Core`.`ModuleInstancePermissionAssigment`
+	(`transaction`, `user`, `moduleInstance`, `permission`, `active`)
+SELECT a.`transaction`, a.`user`, 0, a.`permission`, a.`active`
+FROM `Core`.`SystemModulePermissionAssignment` AS a
+JOIN `Core`.`Uuid` AS uuid
```
Comment (alexander-schmidt) [inline platform-integration/src/main/resources/com/softicar/platform/integration/database/structure/version/v20-migration.sql:10]: I think we need `ON` or `USING` here.

File: platform-integration/src/main/resources/com/softicar/platform/integration/database/structure/version/v20-migration.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+WHERE a.`active`
+AND uuid.`uuidString` = 'a8b076bd-582d-446d-9bce-85a8a180afd5'
```
Comment (alexander-schmidt) [inline platform-integration/src/main/resources/com/softicar/platform/integration/database/structure/version/v20-migration.sql:12]: Missing semicolon.

**Review by alexander-schmidt — CHANGES_REQUESTED** [review]

File: platform-core-module/src/main/java/com/softicar/platform/core/module/access/permission/ConvertedCoreModulePermission.java (hunk 1 above)
Comment (oliver-richers) [reply platform-core-module/src/main/java/com/softicar/platform/core/module/access/permission/ConvertedCoreModulePermission.java:28]: 100%...at least

File: platform-integration/src/main/resources/com/softicar/platform/integration/database/structure/version/v20-migration.sql (hunk 2 above)
Comment (oliver-richers) [reply platform-integration/src/main/resources/com/softicar/platform/integration/database/structure/version/v20-migration.sql:10]: how antiquated this SQL is, pfff

File: platform-core-module/src/main/java/com/softicar/platform/core/module/access/permission/ConvertedCoreModulePermission.java (hunk 1 above)
Comment (oliver-richers) [reply platform-core-module/src/main/java/com/softicar/platform/core/module/access/permission/ConvertedCoreModulePermission.java:28]: actually, this code was not touched, see `EmfConvertedModulePermission.java`

File: platform-integration/src/main/resources/com/softicar/platform/integration/database/structure/version/v20-migration.sql (hunk 3 above)
Comment (oliver-richers) [reply platform-integration/src/main/resources/com/softicar/platform/integration/database/structure/version/v20-migration.sql]: fixed

**Review by alexander-schmidt — APPROVED** [review]

**Pull request merged**